source/game/field/RKGndCol.cc

```diff
@@ -0,0 +1,374 @@
+    u32 nrmOffset = stream.read_u32();
+    u32 prismOffset = stream.read_u32();
+    u32 blockOffset = stream.read_u32();
+
+    assert(posOffset == sizeof(KColHeader));
+
+    m_posData = addOffset(file, posOffset);
+    m_nrmData = addOffset(file, nrmOffset);
+    m_prismData = addOffset(file, prismOffset);
+    m_blockData = addOffset(file, blockOffset);
+
+    m_prismThickness = stream.read_f32();
+    m_areaMinPos.read(stream);
+    m_areaXWidthMask = stream.read_u32();
+    m_areaYWidthMask = stream.read_u32();
+    m_areaZWidthMask = stream.read_u32();
+    m_blockWidthShift = stream.read_u32();
+    m_areaXBlocksShift = stream.read_u32();
+    m_areaXYBlocksShift = stream.read_u32();
+    m_sphereRadius = stream.read_f32();
+
+    m_pos.setZero();
+    m_prevPos.setZero();
+    m_movement.setZero();
+    m_radius = 0.0f;
+    m_prismIter = nullptr;
+    m_cachedPrismArray = m_prismCache.data() - 1;
+    m_prisms = reinterpret_cast<const KCollisionPrism *>(
+            addOffset(m_prismData, sizeof(KCollisionPrism)));
+    m_prismCount =
+            (reinterpret_cast<uintptr_t>(m_blockData) - reinterpret_cast<uintptr_t>(m_prisms + 1)) /
+            sizeof(KCollisionPrism);
+
+    computeBBox();
+}
+
+void RKGndCol::computeBBox() {
+    m_bbox.mMax.set(-999999.0f);
+    m_bbox.mMin.set(999999.0f);
+
+    for (size_t i = 0; i < m_prismCount; i++) {
+        const KCollisionPrism prism = getPrism(i);
+
+        const EGG::Vector3f fnrm = getNrm(prism.fnrm_i);
+        const EGG::Vector3f enrm1 = getNrm(prism.enrm1_i);
+        const EGG::Vector3f enrm2 = getNrm(prism.enrm2_i);
+        const EGG::Vector3f enrm3 = getNrm(prism.enrm3_i);
+        const EGG::Vector3f vtx1 = getPos(prism.pos_i);
+
+        const EGG::Vector3f vtx3 = GetVertex(prism.height, vtx1, fnrm, enrm3, enrm1);
+        const EGG::Vector3f vtx2 = GetVertex(prism.height, vtx1, fnrm, enrm3, enrm2);
+
+        m_bbox.mMin = m_bbox.mMin.minimize(vtx1);
+        m_bbox.mMin = m_bbox.mMin.minimize(vtx2);
+        m_bbox.mMin = m_bbox.mMin.minimize(vtx3);
+        m_bbox.mMax = m_bbox.mMax.maximize(vtx1);
+        m_bbox.mMax = m_bbox.mMax.maximize(vtx2);
+        m_bbox.mMax = m_bbox.mMax.maximize(vtx3);
+    }
+}
+
+EGG::Vector3f RKGndCol::GetVertex(f32 height, const EGG::Vector3f &vertex1,
+        const EGG::Vector3f &fnrm, const EGG::Vector3f &enrm3, const EGG::Vector3f &enrm) {
+    EGG::Vector3f cross = fnrm.cross(enrm);
+    f32 dp = cross.ps_dot(enrm3);
+    cross *= (height / dp);
+
+    return cross + vertex1;
+}
+
+void RKGndCol::lookupSphere(f32 radius, const EGG::Vector3f &pos, const EGG::Vector3f &prevPos,
+        KCLTypeMask typeMask) {
+    f32 radiusClamped = radius;
+    if (radius > this->m_sphereRadius) {
+        radiusClamped = this->m_sphereRadius;
+    }
+    this->m_prismIter = searchBlock(pos);
+    this->m_pos = pos;
+    this->m_prevPos = prevPos;
+    this->m_movement = pos - prevPos;
+    this->m_radius = radiusClamped;
```

Comment:
Regarding `m_radius`, all that's happening here is setting `m_radius` to `std::min(m_sphereRadius, radius)`. Can we reduce to this?

Reply:
Addressed in https://github.com/vabold/Kinoko/pull/43#discussion_r1468789138